source/packet.hpp

```diff
@@ -13,14 +13,16 @@ enum PacketType { ACK, DATA };
 
 struct Packet {
     Packet(PacketType a_type = PacketType::DATA, Size a_size_byte = 0,
-           IFlow* flow = nullptr, Time a_send_time = 0);
+           IFlow* a_flow = nullptr, Id a_source_id = "", Id a_dest_id = "", Time a_RTT = 0, Time a_send_time = 0);
 
     bool operator==(const Packet& packet) const;
     std::string to_string() const;
-    std::shared_ptr<IRoutingDevice> get_destination() const;
 
     std::uint32_t packet_num;
     PacketType type;
+    Id source_id;
+    Id dest_id;
+    Time RTT;
     Size size_byte;
     IFlow* flow;
```

Comment:
As ypu changes pointer type to `Id` in many places, I think this PR is the mest place to to this change:
```suggestion
    Id flow_id;
```

Reply:
Currently desided to keep as it is